.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,49 @@
+{
+  "name": "Jekyll",
```

Comment:
```suggestion
  "name": "Jekyll",
    "runArgs": [
    "--name",
    "cotes2020/jekyll-theme-chirpy"
  ],
```

This defines the container name that will appear in the list of containers and makes it easier to manage them.
Feel free to use some other name per your vision.

Reply:
I agree with naming the container. Since containers are reusable, what are the "naming conflict" scenarios you are worried about and can you please provide more details?

Reply:
Here is the case I was worried about. I may work with both original `cotes2020/jekyll-theme-chirpy` to play with changes from others and my fork `kungfux/jekyll-theme-chirpy` to work on my changes. How this setup will work if the name for container will be defined? That's what I called possible "conflict".

Reply:
This is easily fixed by changing the name of one of the repositories. For example, rename `kungfux/jekyll-theme-chirpy` to `kungfux/my-jekyll-theme-chirpy`.

Imagine if the name of the container was randomly generated, it would be hard to manage because you might not know what the container is about.

Reply:
I found the necessary reason for not specifying the container name:

After we reopen local folder in the container, if we clone the same repo from remote to the container volume at this point it will break with a duplicate name error.

Reply:
I suspect there is a reason to not use `name` prop from `devcontainer.json` to that purpose. So, I'm with you on this one :)

However, I would suggest to clone the repo right into a volume as it makes no sense to clone locally and then clone again remotely.

Reply:
The REMOTE EXPLORER arranges containers by project name, so random container naming doesn't affect our ability to distinguish between projects.

![](https://code.visualstudio.com/assets/docs/devcontainers/containers/containers-explorer.png)

(Source: https://code.visualstudio.com/docs/devcontainers/containers#_managing-containers)

Reply:
That's fine unless you have a bunch of them. I think more on the cleanup containers process from Docker Desktop or Docker CLI. Maybe there is a way to rename the container without affecting devcontainer configuration. It would be great.